lib/class-wp-rest-menu-items-controller.php

```diff
@@ -446,6 +446,7 @@ protected function prepare_item_for_database( $request ) {
 					} else {
 						$prepared_nav_item['menu-item-position'] = count( $menu_items );
 					}
+					array_push($menu_items,$last_item);
 				} else {
```

Comment:
I had to add this in order to be able to save sub-menu items nested under newly-created items. The problem is that the last item is being removed from `$menu_items` on line 443, and `$menu_items` is then used to create a `$menu_item_ids` array that is used to check if an id is a valid parent. The id of the last saved item will never appear in that array because it was popped out of `$menu_items`. 

This solution feels a bit hacky though; ideas for improvement appreciated 🙂